src/tree.js

```diff
@@ -3,6 +3,39 @@ class Tree {
     this.value = value;
     this.children = [];
   }
+  addChild(value) {
+    this.children.push(new Tree(value));
+  }
+  contains(value) {
+    let flag = false;
+    const search = (children) => {
+      for (let i = 0; i < children.length; i++) {
+        if (children[i].value === value) flag = true;
```

Comment:
Same deal here. You'll want to return out of this function once the flag has been set.